site-graph/http/middleware.go

```diff
@@ -0,0 +1,103 @@
+package http
+
+import (
+	"compress/gzip"
+	"crypto/subtle"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	// passwordDigst - hardcoded bcrypt password hash. password: "secret" in plaintext
+	dbPasswordDigst = "$2a$04$QqzenEKqI.CNHTMvH0dPkeQqhLptBURwJSPlKFD0xt1QaQPN/rz26"
+	dbUsername      = "gobridge"
+)
+
+var (
+	ErrUsernameOrPasswordInvalid = fmt.Errorf("username or password is invalid")
+)
+
+// isAuthenticated return false if username or password
+// is not allowed to access application.
+func authenticate(username, password []byte) error {
+	// ConstantTimeCompare returns 1 if the two slices, x and y, have equal contents
+	// and 0 otherwise. The time taken is a function of the length of the slices and
+	// is independent of the contents.
+	if subtle.ConstantTimeCompare(username, []byte(dbUsername)) == 0 {
```

Comment:
nit: for each call you will be converting dbUsername from string to []byte. Maybe worth storing dbUsername as []byte